datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -303,6 +309,21 @@ def http(self):
 
         return self._http
 
+    def get_tls_context(self, refresh=False):
+        # type: (bool) -> ssl.SSLContext
+        """
+        Provides logic to yield consistent network behavior based on user configuration.
+
+        Only new checks or checks on Agent 6.13+ can and should use this for HTTP requests.
+        """
+        if not hasattr(self, '_tls_context_wrapper'):
+            self._tls_context_wrapper = TlsContextWrapper(self.instance or {}, self.TLS_CONFIG_REMAPPER)
+
+        if refresh:
+            self._tls_context_wrapper.refresh_tls_context()
```

Comment:
Let's make the wrapper a `@property` like `http` and manually call a `.refresh()` when needed.

Reply:
That change was intended. I believe that's an important difference between the `http` property and the tls context.
For the RequestsWrapper it made sense to have it as a property because it's used directly by checks to make http calls.
For SSL we use a wrapper class that automates the generation of an SSLContext but what an integration needs simply is the context, having the wrapper has a property complexifies the handling of it from integrations. The reasoning to make `http` a property doesn't really apply for this I believe.